packages/wallet-ui/src/hooks/useHasMetamask.ts

```diff
@@ -36,13 +46,20 @@ export const useHasMetamask = () => {
   };
 };
 
-export const detectMetamask = async () => {
+export const detectMetamask = async (): Promise<EIP6963ProviderDetail | null> => {
+  return new Promise((resolve) => {
+    const handleProvider = (providerDetail: any) => {
+      resolve(providerDetail);
+    };
+    requestProvider(handleProvider);
```

Comment:
the request provider is just addEventListener to "eip6963:announceProvider"

it doesnt handle if it is MM or not